stellargraph/layer/appnp.py

```diff
@@ -149,18 +140,13 @@ def call(self, inputs):
         # Remove singleton batch dimension
         features = K.squeeze(features, 0)
         propagated_features = K.squeeze(propagated_features, 0)
-        out_indices = K.squeeze(out_indices, 0)
 
         # Propagate the node features
         A = As[0]
         output = (1 - self.teleport_probability) * K.dot(
             A, propagated_features
         ) + self.teleport_probability * features
 
-        # On the final layer we gather the nodes referenced by the indices
-        if self.final_layer:
-            output = K.gather(output, out_indices)
-
         # Add batch dimension back if we removed it
         if batch_dim == 1:
```

Comment:
I wonder since we check `batch_dim != 1` and raise `ValueError` in line 135, do we need to check `batch_dim == 1` here again? We would never get to this line if it wasn't true in the first place because of the earlier check.